### PR TITLE
fix: skip checking dirs for required post-analyzers

### LINKED ISSUE #3760: Scans failed & giving "not a valid zip file" error
I think we are loosing basic scan feature here, according to 0.38.0 / 0.38.1
facing issue with python (and or java) based dockers.

the issue remained same irrespective of the option --offline-scan (and / or) --scanners vuln

```
2023-03-03T08:44:43.816Z INFO Vulnerability scanning is enabled
2023-03-03T08:44:43.816Z INFO Secret scanning is enabled
2023-03-03T08:44:43.816Z INFO If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2023-03-03T08:44:43.816Z INFO Please see also https://aquasecurity.github.io/trivy/v0.38/docs/secret/scanning/#recommendation for faster secret detection

2023-03-03T08:44:49.420Z        INFO    JAR files found
2023-03-03T08:44:49.422Z        INFO    Analyzing JAR files takes a while...
2023-03-03T08:44:49.601Z        FATAL   image scan error: scan error: scan failed: failed analysis: analyze error: failed to analyze layer (sha256:125335cc58b0044b35fab83684d58e6ce76dd9f466af964e859fbe7bad8e5eea): post analysis error: post analysis error: walk dir error: on file: jar/war/ear/par parse error: zip error: zip: not a valid zip file
```
can you help in this?

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
There are cases when dir has same name as required file.
e.g.
```
➜  docker run -it --rm  --entrypoint /bin/bash docker.io/consensys/teku:develop-jdk16

teku@6636da39406f:~$ ls -hl ./licenses/
total 604K
drwxr-xr-x 3 teku teku 4.0K Mar  6 05:43 apiguardian-api-1.1.2.jar
drwxr-xr-x 3 teku teku 4.0K Mar  6 05:43 attoparser-2.0.6.RELEASE.jar
drwxr-xr-x 3 teku teku 4.0K Mar  6 05:43 byte-buddy-1.12.21.jar
...
```

We should skip these dirs.

## Related issues
- Close #3760

## Checklist
- [x] I've read the [guidelines for contributing](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/) to this repository.
- [x] I've followed the [conventions](https://aquasecurity.github.io/trivy/latest/community/contribute/pr/#title) in the PR title.
- [ ] I've added tests that prove my fix is effective or that my feature works.
- [ ] I've updated the [documentation](https://github.com/aquasecurity/trivy/blob/main/docs) with the relevant information (if needed).
- [ ] I've added usage information (if the PR introduces new options)
- [ ] I've included a "before" and "after" example to the description (if the PR is a user interface change).

### CODE REVIEW

File: pkg/fanal/analyzer/analyzer.go [hunk 1]
```diff
@@ -456,7 +456,7 @@ func (ag AnalyzerGroup) AnalyzeFile(ctx context.Context, wg *sync.WaitGroup, lim
 func (ag AnalyzerGroup) RequiredPostAnalyzers(filePath string, info os.FileInfo) []Type {
 	var postAnalyzerTypes []Type
 	for _, a := range ag.postAnalyzers {
-		if a.Required(filePath, info) {
+		if !info.IsDir() && a.Required(filePath, info) {
```
[the remark this answers is not in the document]
Comment (DmitriyLewen) [reply pkg/fanal/analyzer/analyzer.go]: Nice catch! Changed in [daaf155](https://github.com/aquasecurity/trivy/pull/3773/commits/daaf155d2c1469df2827e503fa5c7e32d6f4796c)

**Review by knqyf263 — APPROVED** [review]
:shipit:

**Comment by Juliavol** [comment]
Unfortunately version 0.39.0 does not fix https://github.com/aquasecurity/trivy/issues/3760
I am still receiving the same error when using trivu scan in my CI process:
image scan error: scan error: scan failed: failed analysis: analyze error: failed to analyze layer (sha256:dcfc7e5aba4b99abf1efe11040e8984b3c0de76c6ffc835ebd84b9a31fef67c9): post analysis error: post analysis error: walk dir error: on file: "somefolder/mariadb/jdbc/main/mariadb-jdbc.jar" parse error: zip error: zip: not a valid zip file
while iot worked well on v0.37.X

**Comment by DmitriyLewen** [comment]
Hello @Juliavol 
This PR only resolve case when `*.jar` is directory.
For other cases when `*.jar" is not valid - Trivy returns error.

We are working on #3863 to add flag to skip errors.

Regards, Dmitriy

**Pull request merged**